### ISSUE TITLE
Deserializing string with newlines adds extra lines

### ISSUE THREAD

**Issue by superyyrrzz** [aaubry/YamlDotNet#361]
Reproduce code:
```cs
            var obj = new
            {
                Data = @"first line
second line
    third line with indentation
"
            };

            var serializer = new Serializer();
            string yaml = serializer.Serialize(obj);
            var deserializer = new DeserializerBuilder().Build();
            var deserialized = deserializer.Deserialize<Dictionary<string, string>>(yaml);

            Console.WriteLine(deserialized["Data"]);
```

Expect result:
```
first line
second line
    third line with indentation
```

Actual result:
```
first line
second line

    third line with indentation
```

The intermediate yaml is:
```
Data: >+
  first line

  second line

      third line with indentation

```

It seems the default serializer chooses folded scalar style, but it will introduce extra empty line when some line contains indentation. I can workaround this by using `ScalarStyle.Literal` as mentioned in https://github.com/aaubry/YamlDotNet/issues/246#issuecomment-309122787, but I this it's a bug as data is changed after serialization and deseriazation. The serializer may not use folder style as default scalar style.

**Comment by superyyrrzz** [comment]
I find this issue only happens when using `\r\n` line endings. `\n` works well.

**Comment by Shmalikov** [comment]
Hi! 
I've the same issue. I am making my own DSL. It is dynamically resolve yaml files, so I have to use Dictionary<object, object> and can't use attributes to avoid this issue. Is there any progress in fixing this bug?

**Comment by aaubry** [comment]
A fix for this issue has been released in [version 12.0.0](https://github.com/aaubry/YamlDotNet/releases/tag/v12.0.0).
